common/document_parser/lib/section_parse/parsers/utils/dod_utils.py

```diff
@@ -0,0 +1,361 @@
+"""Utilities for DoDParser."""
+
+from re import compile, search, match, IGNORECASE, VERBOSE, Match
+from calendar import month_name
+from typing import Union, Tuple, List, Callable
+
+
+MONTHS = [m for m in month_name[1:]]
+PAGEBREAK_DATE_PATTERN = compile(
+    r"(?:" + r"|".join(MONTHS) + r") [0-9]{1,2},? (?:19[0-9]{2}|2[0-9]{3})"
+)
+
+
+def find_pagebreak_date(text: str) -> Union[Tuple[int, int], None]:
+    """Find the start and end of a pagebreak date within the text.
+
+    Most documents have one date of the following format in each pagebreak:
+        <month> <1-2 digit day> <optional comma> <4-digit year>
+        Example: `March 12, 2004`.
+
+    Args:
+        text (str)
+
+    Returns:
+        Union[Tuple[int, int], None]: If a date is found, returns a Tuple of
+            ints (date start index, date end index). Otherwise, returns None.
+    """
+    date_match = search(PAGEBREAK_DATE_PATTERN, text)
+    if date_match:
+        return (date_match.start(), date_match.end())
+    else:
+        return None
+
+
+def match_alpha_list_item(
+    text: str,
+) -> Union[Tuple[str, Callable], Tuple[None, None]]:
+    """Match an alphabetical (lowercase only) list item.
+
+    Args:
+        text (str)
+
+    Returns:
+        Union[Tuple[str, Callable], Tuple[None, None]]:  If an alphabetical
+            list item is found, the first item of the tuple will be the letter
+            from that item and the second item will be the function that found
+            the match. Otherwise, the both items will be None.
+    """
+    letter = match_alpha_dot(text)
+    if letter:
+        return (letter, match_alpha_dot)
+
+    letter = match_alpha_single_paren(text)
+    if letter:
+        return (letter, match_alpha_single_paren)
+
+    letter = match_alpha_double_parens(text)
+    if letter:
+        return (letter, match_alpha_double_parens)
+
+    return (None, None)
+
+
+def match_alpha_dot(text: str) -> Union[str, None]:
+    """Helper for match_alpha_list_item()."""
+    m = match(r"([a-z])\. ", text)
+    return m.groups()[0] if m else None
+
+
+def match_alpha_single_paren(text: str) -> Union[str, None]:
+    """Helper for match_alpha_list_item()."""
+    m = match(r"([a-z])\) ", text)
+    return m.groups()[0] if m else None
+
+
+def match_alpha_double_parens(text: str) -> Union[str, None]:
+    """Helper for match_alpha_list_item()."""
+    m = match(r"\(([a-z])\) ", text)
+    return m.groups()[0] if m else None
+
+
+def is_sentence_continuation(text: str, prev_text: str) -> bool:
+    """Check if `text` is a continuation of the sentence `prev_text`.
+
+    Note: do NOT strip leading/ trailing whitespace from the params.
+    """
+    if search(r"[a-z] ?[^\.] $", prev_text) and match(r"[a-z]", text):
```

Comment:
Unfortunately, I'm aware of this issue and simple regex logic on its own isn't sufficient because it can cause false positives in other cases.
If this poses a problem downstream, we may want to revisit it as a separate issue.